Author: Roberto Lumbreras <rover@debian.org>
Description: fix arguements spelling error

--- slirp-1.0.17.old/ChangeLog	2004-09-07 05:03:53.000000000 +0200
+++ slirp-1.0.17/ChangeLog	2010-07-04 18:36:26.302051365 +0200
@@ -128,7 +128,7 @@
   so, or if given an argument, every S seconds.
 - When load-balancing over multiple modems, you can now specify the password
   in the environmental variable SLIRP_PASSWORD, then give Slirp the
-  arguements "-l x,host:port,-".  This is for systems where the password is
+  arguments "-l x,host:port,-".  This is for systems where the password is
   still seen in a "ps x" even though Slirp tries to hide it, like Solaris.
 - Add some performance enhancments to the load-balancing code, which
   includes the addition of the "towrite_max N" option, and updated the
--- slirp-1.0.17.old/src/main.c	2006-05-18 21:22:46.000000000 +0200
+++ slirp-1.0.17/src/main.c	2010-07-04 18:36:36.826051981 +0200
@@ -237,7 +237,7 @@
 
 	/* It's an internet socket */
 	if (sscanf(*argv, "%d,%[^:]:%d,%s", &unit, hn, &port, pwd) != 4) {
-	  lprint("Error: bad arguements to -l\n");
+	  lprint("Error: bad arguments to -l\n");
 	  slirp_exit(1);
 	}
 	if (strcmp(pwd, "-") == 0) {
